sycl/include/sycl/detail/common.hpp

```diff
@@ -119,6 +119,72 @@ struct code_location {
 #define _CODELOCFW(a)
 #endif
 
+/// @brief The TLS variable into which Code Locations will be stashed
+extern thread_local detail::code_location GCodeLocTLS;
+
+/// @brief Data type that manages the code_location information in TLS
+/// @details As new SYCL features are added, they all enable the propagation of
+/// the code location information where the SYCL API was called by the
+/// application layer. In order to facilitate this, the tls_code_loc_t object
+/// assists in managing the data in TLS : (1) Populate the information when you
+/// at the top level function in the call chain and (2) to remove the
+/// information when the object goes out scope in the top level function.
+///
+/// Usage:-
+/// void bar() {
+///   tls_code_loc_t p;
+///   // Print the source information of where foo() was called in main()
+///   std::cout << p.query().fileName() << ":" << p.query().lineNumber() <<
+///   std::endl;
+/// }
+/// void bar1() {bar();}
+/// void foo(const code_location &loc) {
+///   tls_code_loc_t tp(loc);
+///   bar1();
+/// }
+/// void main() {
+///   foo(const code_location &loc = code_location::current());
+/// }
```

Comment:
nit: I'd indent the code example by extra two spaces (I.e., everything except "usage" line).

Reply:
:-) Agreed.